pkgs/servers/sql/postgresql/ext/pg-semver.nix

```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  postgresql,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pg-semver";
+  version = "0.40.0";
+
+  buildInputs = [ postgresql ];
+
+  src = fetchFromGitHub {
+    owner = "theory";
+    repo = "pg-semver";
+    rev = "v${version}";
+    hash = "sha256-9f+QuGupjTUK3cQk7DFDrL7MOIwDE9SAUyVZ9RfrdDM=";
+  };
+
+  installPhase = ''
+    install -D -t $out/lib src/semver${postgresql.dlSuffix}
```

Comment:
To be honest, I just started with `pg_uuidv7` and modified it for this new package. The usual `make install` won't work here because it would try to install it into the `postgresql` build rather than `$out`.

Reply:
@Bot-wxt1221 Most other PostgreSQL extensions in this directory use this `installPhase` approach for installing rather than patching because the package's `make install` (actually using `pg_config --pgxs` for the `install` functionality) does not work correctly. I'd be happy to adapt a better way.